modules/swagger-codegen/src/main/resources/TypeScript-Angular/api.mustache

```diff
@@ -0,0 +1,68 @@
+/// <reference path="api.d.ts" />
+
+/* tslint:disable:no-unused-variable member-ordering */
+
+{{#operations}}
+module {{package}} {
+    'use strict';
+
+    {{#description}}
+    /**
+    * {{&description}}
+    */
+    {{/description}}
+    export class {{classname}} {
+        private basePath = '{{basePath}}';
+
+        static $inject: string[] = ['$http'];
+
+        constructor(private $http: ng.IHttpService, basePath?: string) {
+            if (basePath) {
+                this.basePath = basePath;
+            }
+        }
+        {{#operation}}
+        public {{nickname}} ({{#allParams}}{{paramName}}: {{{dataType}}}, {{/allParams}} extraHttpRequestParams?: any ) : ng.IHttpPromise<{{#returnType}}{{{returnType}}}{{/returnType}}{{^returnType}}{}{{/returnType}}> {
+            var path = this.basePath + '{{path}}';
+            {{#pathParams}}
+            path = path.replace('{' + '{{paramName}}' + '}', String({{paramName}}));
+            {{/pathParams}}
+            var queryParameters: any = {};
+            var headers: any = {};
+            {{#allParams}}{{#required}}
+            // verify required parameter '{{paramName}}' is set
+            if (!{{paramName}}) {
+                throw new Error('Missing required parameter {{paramName}} when calling {{nickname}}');
+            }
+            {{/required}}{{/allParams}}
+            {{#queryParams}}if ({{paramName}} !== undefined) {
+                queryParameters['{{paramName}}'] = {{paramName}};
+            }{{/queryParams}}
+            {{#headerParams}}headerParams['{{paramName}}'] = {{paramName}};{{/headerParams}}
```

Comment:
I'm no expert in typescript but should this be `headers['{{paramName}}']` instead of `headerParams['{{paramName}}']` ?